### PR TITLE
Fixes for a bunch of clang-tidy findings

### PR DESCRIPTION
**Pull request by t-8ch**

### CODE REVIEW

File: libmount/src/hook_mount.c [hunk 1]
```diff
@@ -415,7 +415,6 @@ static int set_vfsflags(struct libmnt_context *cxt,
 		rc = api->fd_tree = open_mount_tree(cxt, NULL, (unsigned long) -1);
 		if (rc < 0)
 			return rc;
-		rc = 0;
 	}
```
Comment (karelzak) [inline libmount/src/hook_mount.c:418]: Seems incorrect. The function  open_mount_tree() returns non-zero on success, but the following code assumes rc is 0.

File: libuuid/src/test_uuid.c [hunk 2]
```diff
@@ -66,24 +66,29 @@ static int test_uuid(const char * uuid, int isValid)
...
+		if (str[sz - 1] == '\n')
+			str[sz - 1] = '\0';
 		if (uuid_parse(str, uuidBits)) {
 			warnx("%s: %s", file, str);
 			ret++;
 		}
 	}
 
-	close(fd);
+	fclose(f);
```
Comment (karelzak) [inline libuuid/src/test_uuid.c:91]: maybe also free(str)

File: libmount/src/hook_mount.c (hunk 1 above)
Comment (t-8ch) [reply libmount/src/hook_mount.c:418]: Indeed, dropped.

File: libuuid/src/test_uuid.c (hunk 2 above)
Comment (t-8ch) [reply libuuid/src/test_uuid.c:91]: Done

**Pull request merged**